src/queue.rs

```diff
@@ -65,7 +65,7 @@ impl<T: Send> Live<T> {
 
 /// An idle connection, carrying with it a record of how long it has been idle.
 #[derive(Debug)]
-struct Idle<T: Send> {
+pub(crate) struct Idle<T: Send> {
```

Comment:
I always viewed the `pub(crate)` modifier as only necessary when you want to very finely control what gets public within a single module - x gets to be public but y should only stay within the crate. Since nothing in this module is exposed outside the crate, I don't think the `pub(crate)`s are necessary here, but this is not a blocker.

Reply:
I see.  I actually put those just to explicitly make the function/method is *not* used by the external crate, so that it's free to change, without looking at `lib.rs` file for the module level visibility.  And also, this explicit visibility will help us to avoid the accidental exposure of some internal functions when we change the module level visibility somehow.  I believe we could catch those through the PR process, but once it's out, it's hard to get it back. :)